data/types.go

```diff
@@ -147,3 +160,16 @@ func DefaultExpires(role string) time.Time {
 	}
 	return t.UTC().Round(time.Second)
 }
+
+type unmarshalledSignature Signature
+
+func (s *Signature) UnmarshalJSON(data []byte) error {
+	uSignature := unmarshalledSignature{}
```

Comment:
hmmm, interesting, I haven't come across that before but it seems legit. The other cases of using UnmarshalJSON I've seen appear to be for encoding management rather than object deserialization (specifically data/hex_bytes.go).